hack/testing/test-020-olm-upgrade.sh

```diff
@@ -83,6 +83,10 @@ os::cmd::expect_success "oc -n $NAMESPACE create -f ${repo_dir}/hack/cr.yaml"
 assert_resources_exist
 
 os::cmd::expect_success "deploy_config_map_catalog_source $NAMESPACE ${repo_dir}/manifests '${IMAGE_CLUSTER_LOGGING_OPERATOR}'"
+#patch catalog source
+#olm.skipRange: ">=4.2.0 <4.3.0"
+range=">=$previous_version <$(echo $version | awk '{print $1 + 0.1}')"
+os::cmd::expect_success "oc -n $NAMESPACE get configmap cluster-logging -o yaml | sed -e 's~olm.skipRange:.*~olm.skipRange: ${range}~' | oc replace -n $NAMESPACE -f -"
```

Comment:
```
version=$(basename $(find $manifest_dir -type d | sort -r | head -n 1))
previous_version=$(echo $version | awk '{print $1 - 0.1}')
```
This means previous_version will be `4.2` - This means `range` and `olm.skipRange` will be `>=4.2 <4.3` - is this ok?  Do you have to have a 3 digit version here?  Also note that you could just use `$version` again:
```
range=">=$previous_version <$version"
```

Also - the `os::cmd::expect_success` and others are only really needed for code under test - that is, the actual code you are testing.  For things like deploying the config map source, editing the configmap, etc. - we aren't testing those things, and since we are running the script with `set -e`, the script will report and exit if that command fails.  Finally, using `os::cmd::xx` adds a _lot_ of noise to the output.

Reply:
re: using `$version` in range - never mind - it is good as it is
`range=">=$previous_version <$(echo $version | awk '{print $1 + 0.1}')"`
at least as far as the way you are using version - still don't know if it is ok to use 2 digit versions instead of 3 digit

Reply:
updated to add third digit

Reply:
@richm maybe the missing third digit is the reason this failed

Reply:
maybe - I don't know how it is supposed to work